src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -230,38 +238,46 @@ private async ValueTask<SymbolGroup> ReportGroupAsync(ISymbol symbol, Cancellati
                         _options, cancellationToken).ConfigureAwait(false);
 
                     foreach (var document in foundDocuments)
-                        GetSymbolSet(documentToSymbols, document).Add(symbol);
+                    {
+                        if (!documentToSymbolAndGroup.TryGetValue(document, out var symbolAndGroup))
+                        {
+                            symbolAndGroup = s_symbolToGroupPool.AllocateAndClear();
```

Comment:
Might be mis-reading, but ProcessProjectAsync seems to be running in parallel, is this safe?

Reply:
yup.  that's ok.  this is a different dictionary versus the symbolToGroup  dict created above.  i'll rename this for clarity.

Reply:
Oh I think I misread. got it

Reply:
yeah.  this was unclear.  i renamed this to be more obvious.